boxsdk/auth/jwt_auth.py

```diff
@@ -239,14 +241,32 @@ def _auth_with_jwt(self, sub, sub_type):
         :rtype:
             `unicode`
         """
-        try:
-            return self._construct_and_send_jwt_auth(sub, sub_type)
-        except BoxOAuthException as ex:
-            error_response = ex.network_response
-            box_datetime = self._get_date_header(error_response)
-            if box_datetime is not None and self._was_exp_claim_rejected_due_to_clock_skew(error_response):
-                return self._construct_and_send_jwt_auth(sub, sub_type, box_datetime)
-            raise
+        attempt_number = 0
+        jwt_time = None
+        while True:
+            try:
+                return self._construct_and_send_jwt_auth(sub, sub_type, jwt_time)
+            except BoxOAuthException as ex:
+                jwt_time = None
+                network_response = ex.network_response
+                code = network_response.status_code  # pylint: disable=maybe-no-member
+                box_datetime = self._get_date_header(network_response)
+
+                if attempt_number >= API.MAX_RETRY_ATTEMPTS:
+                    raise ex
+
+                if (code == 429 or code >= 500):
```

Comment:
Lines 259 and 260 are duplicated (262 and 263).  Can you rework this logic so they can appear once?

Reply:
One option is to put the logic in a method. The other is to have a single if statement that checks everything

Reply:
Sure, either works!